### PR TITLE
Feature invcode views

### PR DESCRIPTION
**Pull request by luis-tm**
- Added view for showing the QR code (it will be integrated with the view flow and the API with the story #162
- Added view for scanning the QR code (the actual logic will be implemented with #44)
- Fix to the error handling in create and update lobby
- Modification in the scss of create and uptate lobby to fit labels
- Added dependency react-qr-reader (npm install JodusNodus/react-qr-reader#pull/192/head)

### CODE REVIEW

**Review by mkorob — COMMENTED** [review]
<Route exact path="/lobby/:id/scan/qr">
                    <Header />
                    <ScanQRCode />
                </Route>

I think on this part, shouldn't the link be lobby/scan-qr? (or scan/qr, although I thought a "/" goes for one site , but I am not sure if that's important) - my point is, that I don't think we need a lobby id there? Since we don't first select the lobby to join before scanning the code

**Comment by luis-tm** [comment]
> I think on this part, shouldn't the link be lobby/scan-qr? (or scan/qr, although I thought a "/" goes for one site , but I am not sure if that's important) - my point is, that I don't think we need a lobby id there? Since we don't first select the lobby to join before scanning the code

you're totally right. For scanning it should be without an id

File: src/components/routing/routers/AppRouter.js [hunk 1]
```diff
@@ -38,6 +40,14 @@ const AppRouter = () => {
                     <Header />
                     <LobbyByCode />
                 </Route>
+                <Route exact path="/lobby/:id/share/qr">
```
Comment (paolovic) [inline src/components/routing/routers/AppRouter.js:43]: this starts to become very verbose.
we have already 3 subroutings of lobby
- :id
- :id/share/qr
- scan/qr

I would only reference the lobby routing in the AppRouter, every sub routing in a LobbyRouter

File: src/styles/views/CreateLobby.scss [hunk 2]
```diff
@@ -17,8 +17,8 @@
         color: #FBC12E;
         height: 50px;
         width: 35px;
-        font-size: 1.2em;
-        padding-left: 1em;
+        font-size: 1.0em;
```
Comment (paolovic) [inline src/styles/views/CreateLobby.scss:20]: ah thx!
this fixes the new line I caused by adding "Ranked/Unranked"

File: src/components/views/UpdateLobby.js [hunk 3]
```diff
@@ -66,8 +66,32 @@ const UpdateLobby = props => {
                         headers: { 'token': token }
                     }
                 );
+
             } catch (error) {
-                alert("Something went wrong! ");
+                if (error.response != null) {
+                    // conflict in lobby name
+                    if (error.response.status == 409) {
+                        const popUp = document.getElementById("invalidUser");
```
Comment (paolovic) [inline src/components/views/UpdateLobby.js:74]: do we really have to work on the DOM?
Is there no more elegant solution?

**Review by paolovic — COMMENTED** [review]
Looks good, sorry if I'm too picky

File: src/components/routing/routers/AppRouter.js [hunk 4]
```diff
@@ -38,6 +40,14 @@ const AppRouter = () => {
                     <Header />
```
Comment (paolovic) [inline src/components/routing/routers/AppRouter.js:40]: I don't know about "routing rules", but wouldn't it make sense to rename for example path="/create-lobby", "/public-lobbies" into "/lobby/create" and "/lobbies/public" etc. pp?

File: src/components/views/UpdateLobby.js (hunk 3 above)
Comment (luis-tm) [reply src/components/views/UpdateLobby.js:74]: What happens with this? Why is not elegant?

File: src/components/routing/routers/AppRouter.js (hunk 4 above)
Comment (luis-tm) [reply src/components/routing/routers/AppRouter.js:40]: I agree

**Review by mkorob — APPROVED** [review]
reviewed! :)

**Review by paolovic — APPROVED** [review]

**Pull request merged**